weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/ObjectViewer.tsx

```diff
@@ -499,3 +550,144 @@ const buildBaseRef = (
+      // For the root object, we just want to assign the value to the result
+      if (Array.isArray(result)) {
+        result.push(...value);
+      } else {
+        Object.assign(result, value);
+      }
+    } else {
+      // For all other objects, we want to assign the value to the result
+      context.path.set(result, value);
+    }
+  });
+  return {store, result};
+};
+
+// This function updates the truncatedData from the truncatedStore, adding more data to the truncatedData array, based on the parentID and truncatedCount
+const updateTruncatedDataFromStore = (
+  key: string,
+  truncatedData: Data,
+  truncatedStore: TruncatedStore,
+  truncatedCount: number = ARRAY_TRUNCATION_LENGTH
+) => {
+  const store = {
+    ...truncatedStore,
+  };
+
+  const newData = mapObject(truncatedData, (context: TraverseContext) => {
+    // If the path is the key, we need to show more data
+    if (context.path.toString() === key) {
+      const storeValue = truncatedStore[key].values;
+      // Depending on the length of the store value, we either add truncatedCount more, or the rest of the values
+      if (storeValue.length > truncatedCount) {
+        // Remove the truncated indicator
+        context.value.pop();
+        // Add the new values and truncated indicator
+        context.value.push(...storeValue.slice(0, truncatedCount));
+        context.value.push({
+          [TRUNCATION_KEY]: true,
+        });
+        // Update the store
+        store[key] = {
+          values: storeValue.slice(truncatedCount),
+          index: store[key].index + truncatedCount,
+        };
+      } else {
+        // Remove the truncated indicator
+        context.value.pop();
+        // Add the new values
+        context.value.push(...storeValue);
+        // Update the store
+        delete store[key];
+      }
+    }
+    return context.value;
+  });
+  return {newData, store};
+};
+
+const ShowMoreButtons = ({
+  parentPath,
+  truncatedData,
+  truncatedStore,
+  setTruncatedData,
+  setTruncatedStore,
+}: {
+  parentPath: string;
+  truncatedData: Data;
+  truncatedStore: TruncatedStore;
+  setTruncatedData: (data: Data) => void;
+  setTruncatedStore: (store: TruncatedStore) => void;
+}) => {
+  const truncatedCount = truncatedStore[parentPath]?.values.length ?? 0;
+  return (
+    <Box
+      sx={{
+        display: 'flex',
+        width: '100%',
+        justifyContent: 'flex-end',
+        gap: 1,
+      }}>
+      {truncatedCount > ARRAY_TRUNCATION_LENGTH && (
+        <Button
+          variant="quiet"
+          onClick={() => {
+            const {newData, store} = updateTruncatedDataFromStore(
+              parentPath,
+              truncatedData,
+              truncatedStore,
+              ARRAY_TRUNCATION_LENGTH
+            );
+            setTruncatedData(newData);
+            setTruncatedStore(store);
+          }}>
+          {`Show ${ARRAY_TRUNCATION_LENGTH} more rows`}
+        </Button>
+      )}
+      <Button
```

Comment:
why does this not break the same way? Seems like expanding all would have the bad effect?

Reply:
this does not break because we are not expanding all the children of the array as well, so the render doesnt just instantly break. I felt like forcing users to show more 10 times to access a specific row, was poor UX